include/deal.II/matrix_free/cuda_matrix_free.h

```diff
@@ -507,10 +533,16 @@ namespace CUDAWrappers
     types::global_dof_index *constrained_dofs;
 
     /**
-     *  Mask deciding where constraints are set on a given cell.
+     * Mask deciding where constraints are set on a given cell.
      */
     std::vector<unsigned int *> constraint_mask;
 
+    /**
+     * Vector of boolean, the value i is true if the color i has at least one
+     * cell. It is false otherwise and the color needs to be skipped.
+     */
+    std::vector<bool> color_exists;
```

Comment:
I am not sure if it is really necessary to introduce this vector. Is the information not equivalent to `n_cells[i]>0`?